src/scripts/OSFramework/OSUI/Behaviors/FocusTrap.ts

```diff
@@ -115,15 +120,35 @@ namespace OSFramework.OSUI.Behaviors {
 		}
 
 		// Method to set the focusable elements to be used
-		private _setFocusableElements(): void {
-			this._focusableElements = Helper.Dom.GetFocusableElements(this._targetElement);
-
-			// Check if predicted elements exist at the _focusableElements
-			for (const predictedElement of this._focusableElements.filter(
-				(item) => item === this._predictableTopElement || item === this._predictableBottomElement
-			)) {
-				// If so, remove them from the array collection of _focusableElements
-				this._focusableElements.splice(this._focusableElements.indexOf(predictedElement), 1);
+		private _setFocusableElements(includeTabIndexHidden = false): void {
+			// Ensure the list of focusable elements is empty
+			this._focusableElements = [];
+
+			// Get all the possible focusable
+			const possibleFocusableElements = Helper.Dom.GetFocusableElements(
+				this._targetElement,
+				includeTabIndexHidden
+			);
+
+			// Ensure we get the proper list of focusable elements
+			for (const element of possibleFocusableElements) {
+				// Check if the element is a child of the given targetElement
+				if (
+					element.closest(GlobalEnum.DataBlocksTag.DataBlock) === this._targetElement ||
```

Comment:
Do you think it's worth create a `const` with `element.closest(GlobalEnum.DataBlocksTag.DataBlock)`? Not so much performance, but rather readability of the code.

Reply:
yup, makes sense... ;)

Reply:
Done!